pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/InvalidLogMessageFormatRule.java

```diff
@@ -319,4 +342,50 @@ private int countPlaceholders(final ASTExpression node) {
         }
         return stringLiterals;
     }
+
+    /**
+     * Removes up to {@code maxArgumentsToRemove} arguments from the end of the {@code argumentList},
+     * if the argument is a method call to one of the whitelisted StructuredArguments methods.
+     *
+     * @param maxArgumentsToRemove
+     * @param argumentList
+     */
+    private void removePotentialStructuredArguments(int maxArgumentsToRemove, List<ASTExpression> argumentList) {
+        int removed = 0;
+        while (!argumentList.isEmpty() && removed < maxArgumentsToRemove) {
+            int lastIndex = argumentList.size() - 1;
+            ASTExpression argument = argumentList.get(lastIndex);
+            if (isStructuredArgumentMethodCall(argument)) {
+                argumentList.remove(lastIndex);
+                removed++;
+            } else {
+                // stop if something else is encountered
+                break;
+            }
+        }
+    }
+
+    /*
+     * └─ Expression == argument
+     *    └─ PrimaryExpression
+     *       ├─ PrimaryPrefix
+     *       │  └─ Name: eg. "keyValue" or "StructuredArguments.keyValue"
+     *       └─ PrimarySuffix
+     *          └─ Arguments
+     */
+    private boolean isStructuredArgumentMethodCall(ASTExpression argument) {
+        if (argument.getNumChildren() == 1 && argument.getChild(0) instanceof ASTPrimaryExpression) {
+            ASTPrimaryExpression primary = (ASTPrimaryExpression) argument.getChild(0);
+            if (primary.getNumChildren() == 2 && primary.getChild(1) instanceof ASTPrimarySuffix) {
+                ASTPrimaryPrefix prefix = (ASTPrimaryPrefix) primary.getChild(0);
+                ASTPrimarySuffix suffix = (ASTPrimarySuffix) primary.getChild(1);
+                if (suffix.isArguments() && prefix.getNumChildren() == 1 && prefix.getChild(0) instanceof ASTName) {
+                    ASTName name = (ASTName) prefix.getChild(0);
+                    return name.getImage().startsWith("StructuredArguments.")
+                        || StringUtil.isAnyOf(name.getImage(), STRUCTURED_ARGUMENTS_METHODS);
```

Comment:
This could be replaced by a Set.contains call if `STRUCTURED_ARGUMENTS_METHODS` was a Set